apis/vmware/v1beta1/types.go

```diff
@@ -49,3 +49,8 @@ const (
 	// VirtualMachineStateError is reported if an error occurs determining the status.
 	VirtualMachineStateError = VirtualMachineState("error")
 )
+
+// VirtualMachinePowerOpMode represents the various power operation modes when
+// when powering off or suspending a VM.
+// +kubebuilder:validation:Enum=hard;soft;trySoft
+type VirtualMachinePowerOpMode string
```

Comment:
Sorry last one (I think :)). Let's also add the constants for soft / hard / trySoft

Reply:
VirtualMachinePowerOpMode

Reply:
Yup it is